odoo/addons/base/models/ir_cron.py

```diff
@@ -480,8 +480,14 @@ def _trigger_list(self, at_list):
         if not at_list:
             return
 
-        self.ensure_one()
         now = fields.Datetime.now()
+        self.ensure_one()
+
+        if not self.sudo().active:
+            # skip triggers that would be ignored
+            at_list = [at for at in at_list if at > now]
+            if not at_list:
```

Comment:
Seems weird to have the same early return twice (the two first lines of the method). You can simplify it by

```python
    def _trigger_list(self, at_list):
        """
        Implementation of :meth:`~._trigger`.
        :param list[datetime.datetime] at_list:
            Execute the cron later, at precise moments in time.
        """
        self.ensure_one()
        now = fields.Datetime.now()
        if not self.sudo().active:
              at_list = [at for at in at_list if at > now]

        if not at_list:
              return
        ...
```